core/shared/src/main/scala-2.12+/clock.scala

```diff
@@ -30,6 +31,12 @@ package object clock extends Clock.Service[Clock] {
   final def currentTime(unit: TimeUnit): ZIO[Clock, Nothing, Long] =
     ZIO.accessM(_.clock currentTime unit)
 
+  /**
+   * Get the current time, represented in the current timezone.
+   */
+  def currentDateTime: ZIO[Clock, Nothing, OffsetDateTime] =
+    ZIO.accessM(_.clock.currentDateTime)
+
```

Comment:
@spf3000 I feel that the `timezone` should be accessible on its own, so that I don't have to generate a new timestamp in trying to get it.

I wouldn't really say the information is duplicated. Rather, `timezone` is a primitive from which you construct a higher-order value, `currentDateTime`. The important thing is that one depends on the other.